vendor/github.com/prometheus/prometheus/util/annotations/annotations.go

```diff
@@ -122,6 +122,22 @@ func (e annoErr) Error() string {
 	return fmt.Sprintf("%s (%s)", e.Err, e.PositionRange.StartPosInput(e.Query, 0))
 }
 
+// func ExtractAnnoError(e error) error {
+// 	err, ok := e.(annoErr)
+// 	if !ok {
+// 		return nil
+// 	}
+// 	return err.Err
+// }
+
+func ExtractAnnoError(err error) error {
+	var e annoErr
```

Comment:
can we instead implement `Unwrap() error` on `annonErr`? This way you can do the assertion above just with `errors.Is(anno, annotations.HistogramQuantileForcedMonotonicityInfo)`

Reply:
Good tip, will try, thanks!